src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -286,19 +319,19 @@ default Page<UserDTO> searchAllUsersByLoginOrNameInGroupAndConvertToDTO(Pageable
      */
     @Modifying
     @Transactional // ok because of modifying query
-    @Query("Update User user set user.hideNotificationsUntil = :#{#hideNotificationUntil} where user.id = :#{#userId}")
+    @Query("UPDATE User user SET user.hideNotificationsUntil = :#{#hideNotificationUntil} WHERE user.id = :#{#userId}")
     void updateUserNotificationVisibility(@Param("userId") Long userId, @Param("hideNotificationUntil") ZonedDateTime hideNotificationUntil);
 
     @Modifying
     @Transactional // ok because of modifying query
-    @Query("Update User user set user.langKey = :#{#languageKey} where user.id = :#{#userId}")
+    @Query("UPDATE User user SET user.langKey = :#{#languageKey} WHERE user.id = :#{#userId}")
     void updateUserLanguageKey(@Param("userId") Long userId, @Param("languageKey") String languageKey);
 
     @EntityGraph(type = LOAD, attributePaths = { "groups" })
-    @Query("select user from User user where :#{#groupName} member of user.groups and user not in :#{#ignoredUsers}")
+    @Query("SELECT user FROM User user WHERE user.isDeleted = false AND :#{#groupName} MEMBER OF user.groups AND user NOT IN :#{#ignoredUsers}")
     List<User> findAllInGroupContainingAndNotIn(@Param("groupName") String groupName, @Param("ignoredUsers") Set<User> ignoredUsers);
 
-    @Query("select distinct team.students from Team team where team.exercise.course.id = :#{#courseId} and team.shortName = :#{#teamShortName}")
+    @Query("SELECT DISTINCT team.students FROM Team team WHERE team.exercise.course.id = :#{#courseId} AND team.shortName = :#{#teamShortName}")
```

Comment:
Because this method is only used for authorization checks in scenarios where the deleted user is interacting with the system (attempting to submit an exercise or access a team exercise). Since deleted users can't log in and interact with the system anymore, I skipped this. 

But actually, now I think it wouldn't hurt to filter out just in case another developer uses this function elsewhere assuming it returns non-deleted users.